cartography/intel/azure/__init__.py

```diff
@@ -29,23 +22,24 @@ def _sync_one_subscription(
     neo4j_session: neo4j.Session,
     credentials: Credentials,
     subscription_id: str,
+    requested_syncs: List[str],
     update_tag: int,
     common_job_parameters: Dict,
 ) -> None:
-    compute.sync(neo4j_session, credentials.arm_credentials, subscription_id, update_tag, common_job_parameters)
-    cosmosdb.sync(neo4j_session, credentials.arm_credentials, subscription_id, update_tag, common_job_parameters)
-    aks.sync(neo4j_session, credentials.arm_credentials, subscription_id, update_tag, common_job_parameters)
-    function_app.sync(
-        neo4j_session, credentials.arm_credentials,
-        subscription_id, update_tag, common_job_parameters,
-    )
-    sql.sync(neo4j_session, credentials.arm_credentials, subscription_id, update_tag, common_job_parameters)
-    storage.sync(neo4j_session, credentials.arm_credentials, subscription_id, update_tag, common_job_parameters)
-    key_vaults.sync(neo4j_session, credentials.arm_credentials, subscription_id, update_tag, common_job_parameters)
-    iam.sync(
-        neo4j_session, credentials, credentials.tenant_id, update_tag, common_job_parameters,
-    )
-    network.sync(neo4j_session, credentials.arm_credentials, subscription_id, update_tag, common_job_parameters)
+    for request in requested_syncs:
+        if request in RESOURCE_FUNCTIONS:
+            if request == 'iam':
```

Comment:
this is similar to gcp, we have to pass a different credentials set. which is not a great design :(